components/fxa-client/src/internal/http_client.rs

```diff
@@ -479,10 +485,20 @@ impl Client {
     fn make_oauth_token_request(
         &self,
         config: &Config,
+        session_token: Option<&str>,
         body: serde_json::Value,
     ) -> Result<OAuthTokenResponse> {
         let url = config.token_endpoint()?;
-        Ok(self.make_request(Request::post(url).json(&body))?.json()?)
+        if let Some(session_token) = session_token {
+            let key = derive_auth_key_from_session_token(session_token)?;
+            let request = HawkRequestBuilder::new(Method::Post, url, &key)
+                .body(body)
+                .build()?;
+
+            Ok(self.make_request(request)?.json()?)
```

Comment:
What will be in the `session_token` in the return value in this case?

Reply:
I think that it shouldn't.  In that case, I think we should add a check that `session_token=None`.  I really want to avoid overwriting it here.

Reply:
Yeah that's a good idea, I'm pushing up a change to make it explicit that we prioritize the existing session token